### ISSUE TITLE
Error 403 Forbidden

### ISSUE THREAD

**Issue by Funergy** [marxvn/gdrive#1]
Not working
Getting 403 forbidden on every link I request

**Comment by ibrahimch** [comment]
+1 for this issue

**Comment by alexanderbraga** [comment]
+1 for this issue, Not working, Getting 403 forbidden

**Comment by tuannt065** [comment]
me too, error 403 pls check this issue thanks!

**Comment by palermo33** [comment]
403. That’s an error.

Your client does not have permission to get URL /videoplayback?

help me !! 
iptv6 enable

**Comment by ibrahimch** [comment]
Mail me if your IPv6 Enabled, I have some modefication code, want to check result
pvtlyf(at)gmail.com

**Comment by alochym01** [comment]
me too, error 403 pls check this issue thanks!

**Comment by nxtreaming** [comment]
me too, Error 403.

**Comment by nxtreaming** [comment]
I (nxtreaming at gmail.com) will pay 1000$ for anyone who is the first guy to fix the 403 issue.

The payment is validated before 2017-2-10

**Comment by voxsoftware** [comment]
The solutions is with proxying. I can offer a month plan for use Gdrive back to tunel of my servers

**Comment by nxtreaming** [comment]
@voxsoftware I know proxy server is fine, But I wanna avoiding proxy.

**Comment by voxsoftware** [comment]
there is not current solution, the link only work in the ip that is executing code

**Comment by Logic-Punch** [comment]
Any updates on this?

**Comment by voxsoftware** [comment]
i offer a proxy solution:

Unlimited traffic
Unlimited calls
$5 Monthly per 200Mb bandwith (bandwith no traffic)

**Comment by Funergy** [comment]
Kinda weird it works for me.
And the only changes I made was enable ipv6 in my Digital Ocean droplet.
Use googles DNS and I use cloudflare apart from just installing apache


2017-01-26 2:16 GMT+01:00 voxsoftware <notifications@github.com>:

**Comment by nxtreaming** [comment]
> Kinda weird it works for me.
And the only changes I made was enable ipv6 in my Digital Ocean droplet.
Use googles DNS and I use cloudflare apart from just installing apache

Please verify your server again. 
Google updated the algorithm, and return 403 always.

Does your server still work?

Thanks.

**Comment by Bozz903** [comment]
Who knows how to generate links like that? 
https://lh3.googleusercontent.com/eqMb13ACs-sqQQA3H_3k8K5pZViuV1pZZjCJ7Dhw2zgoEPeh9n0g1WKZa3nE71dkg7whnFkkIg=m18
Please, send me PM

**Comment by ibrahimch** [comment]
@Bozz903 It's possible in google photo only

**Comment by Bozz903** [comment]
@ibrahimch can you please tell how it can be achieved?

**Comment by ibrahimch** [comment]
you can find it in source of google photo's page

**Comment by ibrahimch** [comment]
@Funergy I use same method as you mention above, enable ipv6 in droplet, in resolv file google ipv6 dns, and using cloudflare, It returns on cross IP

**Comment by Funergy** [comment]
Google changed algorithm again 
usually it was
*server*.googlevideo.com
now it is
docs.google.com/

So what you have to do to fix this problem (not the error 403)
is remove the video part from the Regex string in the method where we convert the url to redirector.google.com

**Comment by Bozz903** [comment]
I have working solution, you can contact me via email: infinite9161@gmail.com

**Comment by Logic-Punch** [comment]
@Funergy, does this method still work for you?

My simple setup used to work but then after a day of load it was then
blocked now returning error 403.

**Comment by Funergy** [comment]
Same.
This morning it stopped working.
Still looking as why it stopped. I fixed the above problem that I fixed.
But still 403.
Soooo lets get back to work ;)

2017-01-28 16:46 GMT+01:00 Adynnor <notifications@github.com>:

**Comment by Bozz903** [comment]
I have fresh private fix for this library.
No proxy, no limitation, no cookies.
Contact email: infinite9161@gmail.com

**Comment by voxsoftware** [comment]
With my proxy solution, it's invariable however if Google change your app logic. 
Remember $5 per 200 MB Banwidth, unlimited traffic

**Comment by alochym01** [comment]
there is no updated source code :-1:

**Comment by ibrahimch** [comment]
Any update guys ?

**Comment by docsuleman** [comment]
@Funergy 

it should work on some IP and not others. the issue is not with code. believe me I'm into this field from quite a while. and when it stops working all you have to do is change IP address. You seems digital-ocean user. Ive seen on some movies site that NYC3 VPS of DO with ipv6 is working but not in my case.

**Comment by ibrahimch** [comment]
I've updated the code, premium help available

**Comment by nxtreaming** [comment]
> it should work on some IP and not others. the issue is not with code. believe me I'm into this field from quite a while. and when it stops working all you have to do is change IP address. You seems digital-ocean user. Ive seen on some movies site that NYC3 VPS of DO with ipv6 is working but not in my case.

Yes, It is IP related.

**Comment by Bozz903** [comment]
Sell proxy script for google drive links. Forgot about IP's, IPv6, etc. Price - $50. Free install. infinite9161@gmail.com

**Comment by Bozz903** [comment]
Sell proxy script for google drive links. Forget about IP's, IPv6, etc. Price - $50. Free install help. Moneyback. Tested on CentOS 6,7. Working demo provided by request. infinite9161@gmail.com

**Comment by Funergy** [comment]
@Adynnor You are right on some streaming website I refreshed a couple times and every time the request ip changes.

So the only thing we can do is buy some extra IP's and regularly switch between them.

**Comment by Bozz903** [comment]
Fast proxy for $50. 100mbit/s tested. Have been working 6 months without issues. infinite9161@gmail.com

**Comment by voxsoftware** [comment]
I found a solution For Google Photo, that either have video uploads. I have and API working For this,  contact me For the price. I retrieve the URLs from Google directly without proxy. My email :  dostream.01@gmail.com

**Comment by docsuleman** [comment]
do u have api which can upload 2gb file to google photos? if not then drive to google photo solution is with me and i can share free for community.

**Comment by voxsoftware** [comment]
@docsuleman I have too a platform For streaming  videos, without worry For drive disk space, without worry For file size (up to 5GB)  and without worry about drive maximum plays. If Any interested contact me. I'm using For a friend site.

**Comment by voxsoftware** [comment]
Hello, I just fixed the code for get links from Drive. I offer as service, 10$ monthly. When google updates, I update. Any information to email: dostream.01@gmail.com

**Comment by MediaFireStream** [comment]
Guys....try this. It's open source and you don't have to use 3rd party api.
Demo: http://giftup.tk/mediafire/
